### PR TITLE
search frontend: simplify control flow

### PR DESCRIPTION
**Pull request by rvantonder**

### CODE REVIEW

File: client/shared/src/search/query/decoratedToken.ts [hunk 1]
```diff
@@ -443,7 +443,8 @@ const mapPathMeta = (token: Literal): DecoratedToken[] => {
         if (token.value[current] === '\\') {
             current = current + 2 // Continue past escaped value.
             continue
-        } else if (token.value[current] === '/') {
+        }
+        if (token.value[current] === '/') {
```
Comment (limitedmage) [inline client/shared/src/search/query/decoratedToken.ts:447]: Why though? I think `else if` is fine as it shows that these are mutually exclusive.

Comment (rvantonder) [reply client/shared/src/search/query/decoratedToken.ts:447]: It's simpler. These conditions are actually not mutually exclusive (both can evaluate to false). There's some talk motivating the conversion of else-ifs to ifs and I can't find it right now. We use this convention in the backend, and anecdotally you can actually see that the latter convention is more common for control flow in TS code too:

- [start `if` block after a block that ends in `continue`](https://sourcegraph.com/search?q=repogroup:typescript-gh-100+lang:typescript+%7B...+continue+%7D+if+count:1000&patternType=structural)
- [start `else if` block after a block that ends in `continue`](https://sourcegraph.com/search?q=repogroup:typescript-gh-100+lang:typescript+%7B...+continue+%7D+else+if+count:1000&patternType=structural)

Comment (rvantonder) [reply client/shared/src/search/query/decoratedToken.ts:447]: Sorry--my response was very matter-of-fact. In general it just feels more natural for me to express "do X on condition of Y" independent of "do A on condition of B" when that's the case. A dangling "else if" seems redundant to me and and there's some precedent to support structuring control flow this way (I wrote the original code myself so this is me critiquing me own obvi :-)).

Comment (limitedmage) [reply client/shared/src/search/query/decoratedToken.ts:447]: I see, that makes sense, thanks for the explanation :)

**Review by limitedmage — APPROVED** [review]

**Pull request merged**